processing/src/main/java/org/apache/druid/timeline/partition/PartitionHolder.java

```diff
@@ -33,17 +33,24 @@
 public class PartitionHolder<T extends Overshadowable<T>> implements Iterable<PartitionChunk<T>>
 {
   private final OvershadowableManager<T> overshadowableManager;
+  private short maxMinorVersion;
```

Comment:
I think It's better to keep the concept of maxMinorVersion limited to the timeline class itself. This logic can just live in the `TimelineEntry` class.

Reply:
Please address this comment so that we can merge this PR.
`maxMinorVersion` is not a concept I would want to expose outside of the timeline class. Best to keep all the logic in one place alongwith relevant comments.